### DISCUSSION TITLE
environment variables

### DISCUSSION THREAD

**Discussion by ajoslin103** [npm/feedback#526] [CLI] [answered]
Bringing this over from the old NPX repo as I couldn't find it here.

https://github.com/npm/npx/issues/46

I've tried using env-cmd before running npx, but it's not working...

**What / Why**

Is that possible to support an option -e ( which is short of --env-file)
so that I can put all my environment variables in .env file. and then I can run npx -e webpack or npx -e other.env webpack (default to use .env file, and also you can specify a file)

There is a similar project called dotenv, which allow you to add your runtime environment variables by:

node -r dotenv/config yourscript.js
When you are using a cli library which like webpack, it will become much more complicated. like this:

node -r dotenv/config ./node_modules/.bin/webpack
So I wish we could have it in npx.

**References**

https://www.npmjs.com/package/dotenv

https://github.com/toddbluhm/env-cmd

**Comment by MylesBorins** [comment, marked as answer]
@ajoslin103 if you would like to propose a feature to be added to npx the best next step would be to open an issue over at https://github.com/npm/rfcs to get feedback from the CLI team

**Reply by ajoslin103** [reply]
Done!

thanks!
